src/OrchardCore.Modules/OrchardCore.Deployment/Recipes/LuceneRecipeEventHandler.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using OrchardCore.ContentManagement.Metadata.Records;
+using OrchardCore.Recipes.Events;
+using OrchardCore.Recipes.Models;
+
+namespace OrchardCore.Deployment.Recipes
+{
+    /// <summary>
+    /// This handler provides backward compatibility with ContentIndexSettings that have been migrated to LuceneContentIndexSettings.
+    /// </summary>
+    public class LuceneRecipeEventHandler : IRecipeEventHandler
```

Comment:
@deanmarcussen Yes, the LuceneRecipeEventHandler.RecipeStepExecutingAsync() will be executed on every step. But it will process only on 2 specific steps which are `ReplaceContentDefinition` and `ContentDefinition`. For all the other steps it will just `return Task.CompletedTask;`. The recipeExecutor is responsible for the orders of the steps so it doesn't matter if it passes twice. It will just change the JObject of these steps.

Reply:
@deanmarcussen @Skrypt 

> I vaguely remember you saying recipe event handlers would need to be registered differently

Yes refer to the comments of #6731 and where I added the following

https://github.com/OrchardCMS/OrchardCore/blob/9b015254b5a885617d2bf98987b64612773b1f01/src/OrchardCore/OrchardCore.Recipes.Abstractions/Events/IRecipeEventHandler.cs#L6-L11

Yes, keeping `ContentIndexSettings` doesn' hurt me

But I understand that @Skrypt want to make things cleaner ;)

I will think about it again this week end.

Reply:
We don't need to access the scoped tenant data here, we parse the current recipe.json file that we want to execute. In case of a deployment made from a tenant, the RecipeExecutor will only receive data that will be parsed in the LuceneHandler. In this case, the data is already pre-fetched. So, now there are 2 people that want to keep ContentIndexSettings, I'm about to drop the ball on that PR. I've refactored it twice now. Maybe it's just a bad idea or we just don't agree on the fact that it will make us reach the same goal in the end but differently. Yes, I'm probably stubborn in nature so I prefer having a migration than being obligated to use ContentIndexSettings which is less descriptive. Feel free to refactor it a third time. Sorry, probably had a bad day. But now it makes 3 weeks this PR is up and I won't argue more.